drivers: dts: Add SPI capability to bmi270 driver

dchat-nordic opened the pull request:
Added SPI bus support for BMI270 gyroscope driver.

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>

galak reviewed (dismissed):
Need to update drivers/sensor/bmi270/Kconfig:

```
        select I2C if $(dt_compat_on_bus,$(DT_COMPAT_BOSCH_BMI270),i2c)
        select SPI if $(dt_compat_on_bus,$(DT_COMPAT_BOSCH_BMI270),spi)
```

MaureenHelm reviewed (changes requested):
Please split the renaming/refactoring changes and introduction of SPI capability into two separate commits.

dchat-nordic committed:
sensor: dts: Add SPI capability to bmi270 driver

Added SPI bus support for BMI270 gyroscope driver.

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>

dchat-nordic committed:
sensor: Refactor bmi270 driver data names

Make driver data names consistent in code.

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>

mbolivar-nordic commented on drivers/sensor/bmi270/bmi270.c:
```diff
@@ -584,17 +617,17 @@ static int bmi270_init(const struct device *dev)
-		return -ENODEV;
+	ret = bmi270_bus_check(dev);
+	if (ret < 0) {
+		LOG_ERR("Could not get pointer to %s device", dev->name);
```
Is this really the error you wish to log?

mbolivar-nordic commented on drivers/sensor/bmi270/bmi270.h:
```diff
@@ -1,17 +1,25 @@
 #ifndef ZEPHYR_DRIVERS_SENSOR_BMI270_BMI270_H_
 #define ZEPHYR_DRIVERS_SENSOR_BMI270_BMI270_H_
 
+
```
Remove unnecessary whitespace change

mbolivar-nordic commented on drivers/sensor/bmi270/bmi270.c:
```diff
@@ -545,13 +578,13 @@ static int bmi270_attr_set(const struct device *dev, enum sensor_channel chan,
 	    || (chan == SENSOR_CHAN_ACCEL_XYZ)) {
 		switch (attr) {
 		case SENSOR_ATTR_SAMPLING_FREQUENCY:
-			ret = set_accel_odr_osr(dev, val, NULL);
+			ret = set_accel_odr_osr(val, NULL, dev);
```
I don't think this is a good change.

Having the device pointer first is a common C idiom for emulating object oriented programming. Placing it first is a signal that `dev` is the receiver.

Similarly for the other analogous changes.

dchat-nordic replied:
Ok, I will flip 'dev' to beginning in next push

MaureenHelm replied:
This still needs to be done

mbolivar-nordic commented on drivers/sensor/bmi270/Kconfig:
```diff
@@ -5,8 +5,6 @@
 	bool "BMI270 Inertial measurement unit"
-	default y
-	depends on DT_HAS_BOSCH_BMI270_ENABLED
-	select I2C
+	depends on I2C
```
The point of the PR is adding SPI support for the sensor, but you want to add `depends on I2C`?

mbolivar-nordic reviewed (changes requested):
Overall looks good to me. Some requests.

dchat-nordic commented:
@MaureenHelm @mbolivar-nordic @galak Please review. I added requested changes

MaureenHelm commented on drivers/sensor/bmi270/Kconfig:
```diff
@@ -5,8 +5,7 @@
 
 config BMI270
 	bool "BMI270 Inertial measurement unit"
-	default y
-	depends on DT_HAS_BOSCH_BMI270_ENABLED
```
Don't remove these lines

MaureenHelm reviewed (changes requested):
Please don't change the `struct device *dev` argument position in one commit and then change it back in the next commit. Fixup the original commit instead.

dchat-nordic commented:
@MaureenHelm I fixed the required `dev` position.

gmarull commented on drivers/sensor/bmi270/bmi270.c:
```diff
@@ -18,31 +16,54 @@
 
 LOG_MODULE_REGISTER(bmi270, CONFIG_SENSOR_LOG_LEVEL);
 
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "BMI270 driver enabled without any devices"
+#endif
```
mostly redundant with DT_HAS* helpers

gmarull commented on drivers/sensor/bmi270/bmi270.h:
```diff
@@ -206,8 +213,46 @@ struct bmi270_data {
+struct bmi270_dev_config {
+	union bmi270_bus bus;
+	const struct bmi270_bus_io *bus_io;
 };
```
why is all this in a header file? is it shared with any other module? please move it to the driver otherwise.

gmarull commented on drivers/sensor/bmi270/bmi270_i2c.c:
```diff
@@ -0,0 +1,44 @@
+
+#include "bmi270.h"
+
+#if BMI270_BUS_I2C
```
why not skip compilation of this file if not needed?

mbolivar-nordic replied:
> why not skip compilation of this file if not needed?

That's not possible to do with the current cmake/modules/extensions.cmake, and this is following the pattern for BME280. Unless you have some suggestion for how to do this with existing cmake support for DT?

dchat-nordic replied:
After a talk with @kapi-no he suggested a solution. Please check Kconfig file and CMakeLists.txt to see it.

gmarull commented on drivers/sensor/bmi270/bmi270_spi.c:
```diff
@@ -0,0 +1,104 @@
+#include <zephyr/logging/log.h>
+#include "bmi270.h"
+
+#if BMI270_BUS_SPI
```
ditto. Why isn't that part of the same driver file?

gmarull commented on drivers/sensor/bmi270/bmi270_spi.c:
```diff
@@ -0,0 +1,104 @@
+		.count = ARRAY_SIZE(tx_buf)
+	};
+
+	addr = start & 0x7F;
```
magic constant here

gmarull commented on drivers/sensor/bmi270/bmi270_spi.c:
```diff
@@ -0,0 +1,104 @@
+
+	ret = spi_write_dt(&bus->spi, &tx);
+	if (ret) {
+		LOG_DBG("spi_write_dt failed %i", ret);
```
`LOG_ERR`?

gmarull commented on drivers/sensor/bmi270/bmi270_spi.c:
```diff
@@ -0,0 +1,104 @@
+	addr = start & 0x7F;
+
+	ret = spi_write_dt(&bus->spi, &tx);
+	if (ret) {
```
please use expressions evaluating to boolean, e.g. `ret < 0`

MaureenHelm reviewed (dismissed):
Need to address twister failures, but otherwise LGTM

mbolivar-nordic commented on dts/bindings/sensor/bosch,bmi270-spi.yaml:
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    The BMI270 is an inertial measurment unit. See more info at:
```
You can define this once in bosch,bmi270.yaml and not copy/paste it into the spi and i2c files. Please fix the "measurement" typo.

mbolivar-nordic reviewed (changes requested):
Looks mainly OK for when the merge window opens. Since we have some time until that happens, I'd like to ask for a few more things. Then I am +1.

mbolivar-nordic reviewed (dismissed):
Revoking my -1 since Maureen approved.

dchat-nordic committed:
boards: Remove bmi270 int1-gpios form thingy:53 DTS

BMI270 driver does not support interrupts
and int1-gpios property produces twister failures.

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>
